app/views/shared/_improve_this_page.html.erb

```diff
@@ -0,0 +1,38 @@
+<div class="improve-this-page" data-module="improve-this-page">
+  <div class="js-prompt">
+    <span class="improve-this-page__is-useful-question">Is this page useful?</span>
+    <%= link_to "Yes", contact_govuk_path, class: 'js-page-is-useful' %>
+    <%= link_to "No", contact_govuk_path, class: 'js-offer-feedback' %>
+    <div class="improve-this-page__anything-wrong">
+      <%= link_to "Is there anything wrong with this page?", contact_govuk_path, class: 'js-offer-feedback' %>
+    </div>
+  </div>
+  <div class="js-feedback-form js-hidden">
+    <div class="grid-row">
+      <div class="column-two-thirds">
+        <div class="js-errors"></div>
+        <form>
+          <input type="hidden" name="path" value="<%= request.original_url -%>">
+          <div class="improve-this-page__section">
+            <label>How should we improve this page?</label>
+            <textarea name="description" rows="5"></textarea>
+          </div>
+          <p>Include your name and email address if you'd like us to get back to you.</p>
+          <div class="improve-this-page__section">
+            <label>Name</label>
+            <span>(optional)</span>
```

Comment:
The whole thing is busted for accessibility. I'm hoping to have a chat with Gemma about it.

These labels will need `for` attributes. The yes and no links beside "Is this page useful?" will probably just read out "yes" and "no" and I don't have a good idea how to go about that one. Errors popping up via AJAX sound a little intimidating too, although using the browser validation for the description field will probably be as good as it can be and cover most cases this form is likely to hit.

Redirecting to the contact page, like the JS fallback does, is an option too but I'd much prefer not to.